[firebase] firebase builds can break on macos

### Environment

- Operating System: `Darwin`
- Node Version:     `v20.2.0`
- Nuxt Version:     `0.0.0`
- Nitro Version:    `2.5.2`
- Package Manager:  `pnpm@8.6.3`
- Builder:          `webpack`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://github.com/unjs/nitro-deploys

### Describe the bug

If one dependency uses fsevents, running `NITRO_PRESET=firebase npm run build` will create an un-deployable project.

To fix it, the `.output/server/package.json` and `node_modules` directories needs to be altered manually to remove the `fsevents` dependency, which is macos only and will break cloudbuild. 

This should happen on macos only, windows and linux should be fine.

### Additional context

We can fix it by filtering out fsevents in the firebase preset, since we should be targeting linux and not for darwin, regardless of the environment `firebase deploy` is being run in.

### Logs

```sh
i  functions: creating Node.js 18 (2nd Gen) function server(us-central1)...
Build failed with status: FAILURE and message: npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  undefined
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64

npm ERR! A complete log of this run can be found in:
npm ERR!     /www-data-home/.npm/_logs/2023-07-01T17_56_37_753Z-debug-0.log. For more details see the logs at https://console.cloud.google.com/cloud-build/builds;region=us-central1/3622450c-555e-4490-9d09-9e5dfa875243?project=416452184607.

Functions deploy had errors with the following functions:
        server(us-central1)
i  functions: cleaning up build files...

Error: There was an error deploying functions
```

--- a/src/presets/firebase.ts
+++ b/src/presets/firebase.ts
@@ -72,7 +72,9 @@ async function updatePackageJSON(nitro: Nitro) {
         "firebase-admin": "latest",
         "firebase-functions": "latest",
         ...packageJSON.dependencies,
-      }).sort(([a], [b]) => a[0].localeCompare(b[0]))
+      })
+        .filter((e) => e[0] !== "fsevents")
+        .sort(([a], [b]) => a[0].localeCompare(b[0]))
     ),
     engines: {
       // https://cloud.google.com/functions/docs/concepts/nodejs-runtime

